persistent/tests/test_timestamp.py

```diff
@@ -271,26 +276,37 @@ def test_py_hash_32_64_bit(self):
             py = self._makePy(*self.now_ts_args)
             self.assertEqual(hash(py), bit_32_hash)
 
+
             persistent.timestamp.c_long = ctypes.c_int64
             # call __hash__ directly to avoid interpreter truncation
             # in hash() on 32-bit platforms
-            self.assertEqual(py.__hash__(), bit_64_hash)
+            if not _is_jython:
+                self.assertEqual(py.__hash__(), bit_64_hash)
+            else:
+                # Jython 2.7's ctypes module doesn't properly
+                # implement the 'value' attribute by truncating.
+                # (It does for native calls, but not visibly to Python).
+                # Therefore we get back the full python long. The actual
+                # hash() calls are correct, though, because the JVM uses
+                # 32-bit ints for its hashCode methods.
+                self.assertEqual(py.__hash__(), 384009219096809580920179179233996861765753210540033)
         finally:
             persistent.timestamp.c_long = orig_c_long
 
+        # These are *usually* aliases, but aren't required
+        # to be
```

Comment:
They're not actually aliases on jython. Rather than trying to add more jython specific code, or trying to figure out the bitness of the platform reliably, I fell back on the letter of the law that they're not required to be aliases.